File: win-prog/ProgWinCS/Tapping Into the Keyboard/ExitOnX/ExitOnX.cs
//--------------------------------------
// ExitOnX.cs © 2001 by Charles Petzold
//--------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ExitOnX: Form
{
     public static void Main()
     {
          Application.Run(new ExitOnX());
     }
     public ExitOnX()
     {
          Text = "Exit on X";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;
     }
     protected override void OnKeyDown(KeyEventArgs kea)
     {
          if (kea.KeyCode == Keys.X)
               Close();
     }
}

File: win-prog/ProgWinVB/Tapping Into the Keyboard/ExitOnX/ExitOnX.vb
'----------------------------------------
' ExitOnX.vb (c) 2002 by Charles Petzold
'----------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ExitOnX
    Inherits Form

    Shared Sub Main()
        Application.Run(new ExitOnX())
    End Sub

    Sub New()
        Text = "Exit on X"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText
    End Sub

    Protected Overrides Sub OnKeyDown(ByVal kea As KeyEventArgs)
        If kea.KeyCode = Keys.X Then Close()
    End Sub
End Class
